compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/TailrecLowering.kt

```diff
@@ -133,28 +136,31 @@ private class BodyTransformer(
         val specifiedParameters = parameterToArgument.map { (parameter, _) -> parameter }.toSet()
 
         // For each unspecified argument set the corresponding variable to default:
-        parameters.filter { it !in specifiedParameters }.forEach { parameter ->
-
-            val originalDefaultValue = parameter.defaultValue?.expression ?: throw Error("no argument specified for $parameter")
-
-            // Copy default value, mapping parameters to variables containing freshly computed arguments:
-            val defaultValue = originalDefaultValue
-                .deepCopyWithVariables()
-                .transform(object : IrElementTransformerVoid() {
-
-                    override fun visitGetValue(expression: IrGetValue): IrExpression {
-                        expression.transformChildrenVoid(this)
-
-                        val variable = parameterToVariable[expression.symbol.owner] ?: return expression
-                        return IrGetValueImpl(
-                            expression.startOffset, expression.endOffset, variable.type,
-                            variable.symbol, expression.origin
-                        )
-                    }
-                }, data = null)
-
-            +irSetVar(parameterToVariable[parameter]!!.symbol, defaultValue)
-        }
+        parameters
+            .filter { it !in specifiedParameters }
+            .let { if (properComputationOrderOfTailrecDefaultParameters) it else it.asReversed() }
```

Comment:
The addition of this extra let is the only change here, the rest is just formatting.